### PR TITLE
[CORE] Add more units

### PR DESCRIPTION
**Pull request by VonOx**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- ~~[ ] If your changes affects code, did your write the tests?~~
- [X] Are tests passing? (`npm test` on both front/server)
- [X] Is the linter passing? (`npm run eslint` on both front/server)
- [X] Did you run prettier? (`npm run prettier` on both front/server)
- [X] If you are adding a new features/services, did you run integration comparator? (`npm run compare-translations` on front)
- ~~[ ] Did you test this pull request in real life? With real devices? If this development is a big feature or a new service, we recommend that you provide a Docker image to [the community](https://community.gladysassistant.com/) for testing before merging.~~
- ~~[ ] If your changes modify the API (REST or Node.js), did you modify the API documentation? (Documentation is based on comments in code)~~
- ~~[ ] If you are adding a new features/services which needs explanation, did you modify the user documentation? See [the GitHub repo](https://github.com/GladysAssistant/v4-website) and the [website](https://gladysassistant.com).~~
- ~~[ ] Did you add fake requests data for the demo mode (`front/src/config/demo.js`) so that the demo website is working without a backend? (if needed) See [https://demo.gladysassistant.com](https://demo.gladysassistant.com).~~

NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open.

### Description of change

Just adding some units asked by community and needed for future integrations ( Z2M Lixee TIC 👀 )

### CODE REVIEW

**Review by atrovato — DISMISSED** [review]

File: front/src/config/i18n/en.json [hunk 1]
```diff
@@ -1556,36 +1556,76 @@
   "deviceFeatureUnit": {
```
Comment (Pierre-Gilles) [inline front/src/config/i18n/en.json:1556]: In this file, the `deviceFeatureCategory` attribute should be filled for new category

File: front/src/config/i18n/fr.json [hunk 2]
```diff
@@ -1556,36 +1556,76 @@
   "deviceFeatureUnit": {
```
Comment (Pierre-Gilles) [inline front/src/config/i18n/fr.json:1556]: Same in this file !

**Review by Pierre-Gilles — CHANGES_REQUESTED** [review]
Nice PR, thanks @VonOx ! 

There are some translations missing in the i18n files, so those attributes could be used in the UI (in the MQTT integration for example)

File: front/src/config/i18n/en.json (hunk 1 above)
Comment (Pierre-Gilles) [reply front/src/config/i18n/en.json:1556]: https://github.com/GladysAssistant/Gladys/blob/master/front/src/config/i18n/en.json#L1608

**Review by VonOx — COMMENTED** [review]
@Pierre-Gilles deviceFeatureCategory is already filled 😕 

image

**Comment by Pierre-Gilles** [comment]
@VonOx Weird, because I'm not able to use those devices in the MQTT integration. Is it working on your side ?

**Comment by VonOx** [comment]
I miss some mapping in front/server, it's now better

**Comment by VonOx** [comment]
@Pierre-Gilles Can you look again please ?

**Comment by Terdious** [comment]
Hello @VonOx,
To see how to turn the thing, but in connection with the heating, could you add a category "Time" which will be a command and return of state. The goal is to be able to display it either in date+hour+minute or in duration (declared in minutes for example) if > 60 then hours are displayed, otherwise minutes.
So for types:
- a duration in integer I think (the goal here being to be able to display the remaining duration, to increase it or decrease it)
- dateTimeUTC (format 1651346855 for example - the goal here being to display the end date without possible command? or add this one in the time-sensor)
No extra unit I guess.
Tell me what you think of it.
to be able to display something like this:
image

**Comment by VonOx** [comment]
Hi @Terdious 

* The first one is already added ( I think ? ) , but duration seems to be a better term 
```json
"time-sensor": {
      "shortCategoryName": "Time sensor",
      "decimal": "Time (decimal)",
      "integer": "Time (integer)"
    }
```
Combined with unit of your choice

```json
{
    "microseconds": "μs",
    "milliseconds": "ms",
    "seconds": "s",
    "minutes": "min",
    "hours": "h",
    "days": "j",
    "weeks": "sem",
    "months": "m",
    "years": "an."
}
```
* The second one must be calculated I think (dayjs will do the math) , there is no device feature for that ?

File: front/src/config/i18n/fr.json [hunk 3]
```diff
@@ -1760,6 +1828,41 @@
...
+      "index": "Index"
+    },
+    "volume-sensor": {
+      "shortCategoryName": "Capteur de volume",
+      "decimal": "Volume (décimale)",
+      "integer": "Volume (entier)"
+    },
+    "currency": {
+      "shortCategoryName": "Monnaie",
+      "cost": "Coût"
```
Comment (Pierre-Gilles) [inline front/src/config/i18n/fr.json]: "Coût" is weird, I tried to use this feature without looking at this file, and I couldn't find it. 

Maybe the category should be like "Bourse" ou "Devise", with inside "Monnaie" 

Looks like this now:

Screenshot 2022-05-03 at 09 51 47

Comment (Pierre-Gilles) [reply front/src/config/i18n/fr.json]: Also, it should be a decimal

File: front/src/config/i18n/fr.json [hunk 4]
```diff
@@ -1760,6 +1828,41 @@
...
+    },
+    "precipitation-sensor": {
+      "shortCategoryName": "Capteur de précipitations",
+      "decimal": "Précipitation"
+    },
+    "uv-sensor": {
+      "shortCategoryName": "Capteur d'UV",
+      "integer": "Indice UV"
+    },
+    "time-sensor": {
```
Comment (Pierre-Gilles) [inline front/src/config/i18n/fr.json]: "time-sensor" is weird, duration is better!

File: front/src/config/i18n/en.json [hunk 5]
```diff
@@ -1760,6 +1828,41 @@
...
+      "index": "Index"
+    },
+    "volume-sensor": {
+      "shortCategoryName": "Volume sensor",
+      "decimal": "Volume (decimal)",
+      "integer": "Volume (integer)"
+    },
+    "currency": {
+      "shortCategoryName": "Currency",
+      "cost": "Cost"
```
Comment (Pierre-Gilles) [inline front/src/config/i18n/en.json]: Same here, should be decimal

"Trading" or "Money" as category name

and inside "Currency"

File: front/src/config/i18n/fr.json [hunk 6]
```diff
@@ -1556,36 +1556,104 @@
...
+    "uv-index": "Indice UV",
+    "microseconds": "μs",
+    "milliseconds": "ms",
+    "seconds": "s",
+    "minutes": "min",
+    "hours": "h",
+    "days": "j",
+    "weeks": "sem",
+    "months": "m",
+    "years": "an."
```
Comment (Pierre-Gilles) [inline front/src/config/i18n/fr.json]: why the "." ?

File: front/src/config/i18n/fr.json [hunk 7]
```diff
@@ -1760,6 +1828,41 @@
       "stop": "Stop",
       "record": "Enregistrer"
     },
+    "energy-sensor": {
+      "shortCategoryName": "Capteur d'énergie",
```
Comment (Pierre-Gilles) [inline front/src/config/i18n/fr.json:1832]: capteur d'énergie? c'est quoi ?

**Review by Pierre-Gilles — CHANGES_REQUESTED** [review]
Great PR! It'll really help users :) 

I have a few changes.

Don't hesitate to open Gladys when doing this, then going to the MQTT integration, and see how it's looking in the UI, it should be easily searchable and understandable.

**Comment by Pierre-Gilles** [comment]
@Terdious I don't think we want to add this in this PR. It looks like there is some work to think about on heating devices, and if we include this development in this PR, this PR will never go out now!

File: front/src/config/i18n/fr.json (hunk 7 above)
Comment (VonOx) [reply front/src/config/i18n/fr.json:1832]: Des devices spécifiques à la mesure de l'énergie

Exemple qui arrive https://www.zigbee2mqtt.io/devices/ZLinky_TIC.htmle

File: front/src/config/i18n/fr.json (hunk 6 above)
Comment (VonOx) [reply front/src/config/i18n/fr.json]: C'est l'abréviation de année(s)

File: front/src/config/i18n/fr.json (hunk 4 above)
Comment (VonOx) [reply front/src/config/i18n/fr.json]: I agree

**Review by Pierre-Gilles — APPROVED** [review]
Ca me parait bien !

**Pull request merged**